Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2837,8 +2837,8 @@ extension BrowserViewController: ContextMenuHelperDelegate {
                     }
                     accessDenied.addAction(settingsAction)
                     self.present(accessDenied, animated: true, completion: nil)
-                    LeanplumIntegration.sharedInstance.track(eventName: .saveImage)
                 }
+                LeanplumIntegration.sharedInstance.track(eventName: .saveImage)
```

Comment:
This should not be here. This should be under the `UIImageWriteToSavedPhotosAlbum` on line 2830. At that point the user has actually saved an image to the album.